### PR TITLE
Add Azure client and improve AWS error handling

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#3377

Needed for Azure support for the cluster state e2e test I'm adding to `e2etests`. This is so the master node can be rebooted.

I also updated the error handling to check the necessary env vars have been set.

### CODE REVIEW

File: azure/azure.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+var (
+	azureClientID       string
+	azureClientSecret   string
+	azureSubscriptionID string
+	azureTenantID       string
+)
+
+type Client struct {
+	VirtualMachineClient *compute.VirtualMachinesClient
```
Comment (rossf7) [inline azure/azure.go:28]: This is the client for the reboot API call.

File: aws/aws.go [hunk 2]
```diff
@@ -8,37 +8,87 @@ import (
...
 	EC2            *ec2.EC2
 	CloudFormation *cloudformation.CloudFormation
 }
 
-func NewClient() *Client {
+func NewClient() (*Client, error) {
 	a := &Client{}
 
 	{
+		region = os.Getenv(envVarRegion)
```
Comment (fgimenez) [inline aws/aws.go:43]: Cool! <3 I was thinking of having a lower level check on e2e-harness were `project.yaml` is parsed, but at this level is great.

**Review by fgimenez — APPROVED** [review]
Very nice, LGTM

File: aws/aws.go (hunk 2 above)
Comment (rossf7) [reply aws/aws.go:43]: Thanks! It would be useful to have checks in e2e-harness too. But I think its useful to check here too.

**Pull request merged**